src/taco-utils/tacoPackageLoader.ts

```diff
@@ -96,6 +98,38 @@ module TacoUtility {
 
         public static mockForTests: TacoUtility.ITacoPackageLoader;
 
+        public static lazyRun(packageName: string, packageId: string, commandName: string): Q.Promise<string> {
+            var request: IPackageInstallRequest = TacoPackageLoader.createPackageInstallRequest(packageName, packageId, InstallLogLevel.warn);
+
+            return Q({})
+                .then(function (): void {
+                    TacoPackageLoader.installPackageIfNeeded(request);
+                })
+                .then(function (): Q.Promise<string> {
+                    var packageJsonFilePath = path.join(request.targetPath, "package.json");
+                    var packageJson = JSON.parse(<any> fs.readFileSync(packageJsonFilePath));
+
+                    if (!packageJson.bin) {
+                        return Q.reject<string>(errorHelper.get(TacoErrorCodes.PackageLoaderRunPackageDoesntExportBinary, packageName));
+                    }
+
+                    if (!packageJson.bin[commandName]) {
+                        return Q.reject<string>(errorHelper.get(TacoErrorCodes.PackageLoaderRunPackageDoesntHaveRequestedBinary, packageName, commandName));
+                    }
+
+                    var commandFilePath = path.join(request.targetPath, packageJson.bin[commandName]);
```

Comment:
Is this logic correct? My understanding was that if a package specified `bin: {"foo": "./bar.js"}` then it would create a node_modules/.bin/foo which was appropriately linked (foo.cmd on windows, symlinked with executable permissions on Linux/mac) to the ./bar.js file. With this logic, won't you be pointing to the .js file and try to run that, which will cause issues on windows?